x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -392,6 +395,30 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     return !!(scalingType === SCALING_TYPES.TOP_HITS && topHitsSplitField);
   }
 
+  async isEditable(): Promise<boolean> {
+    if (!getMapAppConfig().enableDrawingFeature) {
+      this._isEditable = false;
+      return this._isEditable;
+    }
+    if (this._isEditable === undefined) {
+      if (!this.indexPattern) {
+        await this.getIndexPattern();
+      }
+
+      const { matchingIndexes } = await getMatchingIndexes(this.indexPattern!.title);
+      this._isEditable =
+        // For now we only support 1:1 index-pattern:index matches
+        matchingIndexes.length === 1 &&
+        // Permissions required for index modification are identical to file upload
+        (await getFileUpload().hasImportPermission({
+          checkCreateIndexPattern: true,
```

Comment:
Do you need to check that user can create index pattern? Since the index pattern already exists, why is this permission important?